lang/java/reef-io/src/test/java/org/apache/reef/services/network/DefaultNetworkServiceTest.java

```diff
@@ -0,0 +1,746 @@
+    final IdentifierFactory factory = injectorReceiver.getNamedInstance(NetworkServiceParameters.NetworkServiceIdentifierFactory.class);
+    receiverNetworkService.registerId(factory.getNewInstance(receiver));
+    receiverNetworkService.registerConnectionFactory(groupCommClientId, new StringCodec(), new MessageHandler<String>(monitor, groupcommMessages), new TestListener<String>());
+    receiverNetworkService.registerConnectionFactory(shuffleClientId, new ObjectSerializableCodec<Integer>(), new MessageHandler<Integer>(monitor2, shuffleMessges), new TestListener<Integer>());
+
+    // network service for sender
+    LOG.log(Level.FINEST, "=== Test network service sender start");
+    final String sender = "sender";
+    final Injector injectorSender = injector.forkInjector(netConf);
+    final NetworkServiceClient senderNetworkService = injectorSender.getInstance(NetworkServiceClient.class);
+    senderNetworkService.registerId(factory.getNewInstance(sender));
+    senderNetworkService.registerConnectionFactory(groupCommClientId, new StringCodec(), new MessageHandler<String>(monitor, groupcommMessages), new TestListener<String>());
+    senderNetworkService.registerConnectionFactory(shuffleClientId, new ObjectSerializableCodec<Integer>(), new MessageHandler<Integer>(monitor2, shuffleMessges), new TestListener<Integer>());
+
+    // connection for sending messages
+    final ConnectionFactory<String> gcSender = senderNetworkService.getConnectionFactory(groupCommClientId);
+    final ConnectionFactory<Integer> shuffleSender = senderNetworkService.getConnectionFactory(shuffleClientId);
+
+    final Identifier destId = factory.getNewInstance(receiver);
+    final Connection<String> gcConnectionToReceiver = gcSender.newConnection(destId);
+    final Connection<Integer> shuffleConnectionToReceiver = shuffleSender.newConnection(destId);
+    try {
+      gcConnectionToReceiver.open();
+      shuffleConnectionToReceiver.open();
+
+      executor.submit(new Runnable() {
+        @Override
+        public void run() {
+          for (int count = 0; count < groupcommMessages; ++count) {
+            gcConnectionToReceiver.write("hello! " + count);
+          }
+        }
+      });
+
+      executor.submit(new Runnable() {
+        @Override
+        public void run() {
+          for (int count = 0; count < shuffleMessges; ++count) {
+            shuffleConnectionToReceiver.write(count);
+          }
+        }
+      });
+
+      monitor.mwait();
+      monitor2.mwait();
+
+    } catch (NetworkException e) {
+      e.printStackTrace();
+    }
+    gcConnectionToReceiver.close();
+    shuffleConnectionToReceiver.close();
+
+    senderNetworkService.close();
+    receiverNetworkService.close();
+
+    executor.shutdown();
+    nameServer.close();
+  }
+
+  /**
+   * Test NetworkService registering multiple connection factories which contain streamingcodec.
+   */
+  @Test
+  public void testMultipleConnectionFactoriesStreamingTest() throws Exception {
```

Comment:
Can you refactor testMultipleConnectionFactoriesStreamingTest?
There's duplicate code in testMultipleConnectionFactoriesTest and testMultipleConnectionFactoriesStreamingTest.